lib/bmp280/calibration/bmp180_calibration.ex

```diff
@@ -0,0 +1,91 @@
+defmodule BMP280.BMP180Calibration do
+  @moduledoc false
+
+  @type t() :: %{
+          type: :bmp180,
+          ac1: integer,
+          ac2: integer,
+          ac3: integer,
+          ac4: integer,
+          ac5: integer,
+          ac6: integer,
+          b1: integer,
+          b2: integer,
+          mb: integer,
+          mc: integer,
+          md: integer
+        }
+
+  @two_2 :math.pow(2, 2)
+  @two_4 :math.pow(2, 4)
+  @two_8 :math.pow(2, 8)
+  @two_11 :math.pow(2, 11)
+  @two_12 :math.pow(2, 12)
+  @two_13 :math.pow(2, 13)
+  @two_15 :math.pow(2, 15)
+  @two_16 :math.pow(2, 16)
+
+  @spec from_binary(binary) :: t()
+  def from_binary(
+        <<ac1::16, ac2::16, ac3::16, ac4::16, ac5::16, ac6::16, b1::16, b2::16, mb::16, mc::16,
+          md::16>>
```

Comment:
Please see if the following works instead:

```elixir
<<ac1::big-signed-16, ac2::big-signed-16, ac3::big-signed-16, ac4::big-unsigned-16,
  ac5::big-unsigned-16, ac6::big-unsigned-16, b1::big-signed-16, b2::big-signed-16,
  mb::big-signed-16, mc::big-signed-16, md::big-signed-16>>
```
or
```elixir
<<ac1::signed-16, ac2::signed-16, ac3::signed-16, ac4::unsigned-16,
  ac5::unsigned-16, ac6::unsigned-16, b1::signed-16, b2::signed-16,
  mb::signed-16, mc::signed-16, md::signed-16>>
```

Then if that is the case the swapping functions will not be necessary.

Reply:
I had problems to initially get it working but this works ok, thanks